hack/vmimage-validate.sh

```diff
@@ -14,15 +12,17 @@ trap cleanup EXIT
 
 if [ -z "$IMAGE_RESOURCENAME" ] ;
 then
-  IMAGE_RESOURCENAME=$(az image list -g images --query '[-1].name')
+  IMAGE_RESOURCENAME=$(az image list -g images --query '[-1].name' -o tsv)
 fi
 
+BUILD_RESOURCE_GROUP="vmimage-$(date +%Y%m%d%H%M)"
 IMAGE_RESOURCEGROUP="${IMAGE_RESOURCEGROUP:-images}"
 IMAGE_STORAGEACCOUNT="${IMAGE_STORAGEACCOUNT:-openshiftimages}"
 
 echo "Validating: ${IMAGE_RESOURCENAME}"
 
-go generate ./... && go run -ldflags "-X main.gitCommit=$(git rev-parse --short=10 HEAD)" ./cmd/vmimage -imageResourceGroup "${IMAGE_RESOURCEGROUP}" -image "${IMAGE_RESOURCENAME}" -imageStorageAccount "${IMAGE_STORAGEACCOUNT}" -validate "true"
+# pass -preserveBuildResourceGroup so we can copy artifacts
+go generate ./... && go run -ldflags "-X main.gitCommit=$(git rev-parse --short=10 HEAD)" ./cmd/vmimage -buildResourceGroup "${BUILD_RESOURCE_GROUP}" -imageResourceGroup "${IMAGE_RESOURCEGROUP}" -image "${IMAGE_RESOURCENAME}" -imageStorageAccount "${IMAGE_STORAGEACCOUNT}" -validate -preserveBuildResourceGroup
 
 # Currently there are only 3 logs we want to capture
-mv /tmp/{yum_update_info,yum_check_update,scap-report.html} ${ARTIFACTS:-/tmp}/
+mv /tmp/{yum_update_info,yum_check_update,scap-report.html} ${ARTIFACTS:-/tmp}/ || true
```

Comment:
When would this fail?

Reply:
I guess if one of those files `yum_update_info` or `yum_check_update` or `scap-report.html` isn't available in `/tmp` then mv will return a non-zero exit code. @kwoodson  I wonder if that is a possibility with the report generation...that one of those files might not get written for some reason?

Reply:
I think, it happens when `ARTIFACTS` is not set or set to `/tmp` so the command will basically be moving files to the same location:

```
mv /tmp/{yum_update_info,yum_check_update,scap-report.html} /tmp/
mv: '/tmp/yum_update_info' and '/tmp/yum_update_info' are the same file
mv: '/tmp/yum_check_update' and '/tmp/yum_check_update' are the same file
mv: '/tmp/scap-report.html' and '/tmp/scap-report.html' are the same file

echo $?
1
```

Reply:
@m1kola beat me to it!

@ehashman @charlesakalugwu @mjudeikis 
The major reason I placed the `|| true` here is that when running locally and the `$ARTIFACTS` variable is unset and defaults to tmp.  The `mv` essentially moves the file to the same location. `mv: '/tmp/scap-report.html' and '/tmp/scap-report.html' are the same file`.

I did not want to fail the entire process because the file gets moved to the same location.  If there is a preferred method to mitigate this then I'm open for suggestions.